src/Compilers/CSharp/Portable/Parser/DirectiveParser.cs

```diff
@@ -681,6 +685,26 @@ private DirectiveTriviaSyntax ParseShebangDirective(SyntaxToken hash, SyntaxToke
             return SyntaxFactory.ShebangDirectiveTrivia(hash, exclamation, this.ParseEndOfDirectiveWithOptionalPreprocessingMessage(), isActive);
         }
 
+        private DirectiveTriviaSyntax ParseIgnoredDirective(SyntaxToken hash, SyntaxToken colon, bool isActive, bool isFollowingToken)
+        {
+            if (!lexer.Options.FileBasedProgram)
+            {
+                colon = this.AddError(colon, ErrorCode.ERR_PPIgnoredNeedsFileBasedProgram);
+            }
+
+            if (isFollowingToken)
+            {
+                colon = this.AddError(colon, ErrorCode.ERR_PPIgnoredFollowsToken);
+            }
+
+            if (_context.SeenAnyIfs)
+            {
+                colon = this.AddError(colon, ErrorCode.ERR_PPIgnoredFollowsIf);
+            }
+
+            return SyntaxFactory.IgnoredDirectiveTrivia(hash, colon, this.ParseEndOfDirectiveWithOptionalPreprocessingMessage(), isActive);
```

Comment:
oh interesting.  the pp message isn't something you can grab as a direct text token?  i don't lolve that.  but i guess it could work.

Reply:
Agreed, I feel like this should be a direct token.

Reply:
Well, I made this  consistent with the shebang directive, but I can change it.